### PR TITLE
[core/base] clang-tidy modernize nullptr

### PR DESCRIPTION
**Pull request by ferdymercury**
As suggested in a previous PR, here a smaller PR for nullptr modernization, on the already tidied up core/base (https://github.com/root-project/root/issues/7412 thanks @linev )

### CODE REVIEW

File: core/base/src/TFileCollection.cxx [hunk 1]
```diff
@@ -770,7 +770,7 @@ TObjString *TFileCollection::ExportInfo(const char *name, Int_t popt)
 
 TFileCollection *TFileCollection::GetFilesOnServer(const char *server)
 {
-   TFileCollection *fc = (TFileCollection *)0;
+   TFileCollection *fc = (TFileCollection *)nullptr;
```
Comment (ferdymercury) [inline core/base/src/TFileCollection.cxx:773]: the cast is probably not needed?

Comment (pcanal) [reply core/base/src/TFileCollection.cxx:773]: I agree.

**Comment by pcanal** [comment]
@phsft-bot build

**Comment by linev** [comment]
@phsft-bot build

**Comment by pcanal** [comment]
@phsft-bot build

**Comment by pcanal** [comment]
@phsft-bot build

**Comment by pcanal** [comment]
@ferdymercury sorry for the delay.  Would you be able to rebase onto master, apparently the automated merge is failing.  Thanks.

**Comment by ferdymercury** [comment]
Closing in favour of https://github.com/root-project/root/pull/8045

**Pull request closed without merging**